pinot-core/src/main/java/org/apache/pinot/core/util/SegmentCompletionProtocolUtils.java

```diff
@@ -0,0 +1,103 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.util;
+
+import java.io.File;
+import java.net.URI;
+import org.apache.pinot.common.metrics.ServerMeter;
+import org.apache.pinot.common.metrics.ServerMetrics;
+import org.apache.pinot.common.protocols.SegmentCompletionProtocol;
+import org.apache.pinot.common.utils.FileUploadDownloadClient;
+import org.apache.pinot.server.realtime.ControllerLeaderLocator;
+import org.slf4j.Logger;
+
+
+/**
+ * Util methods related to low level consumers' segment completion protocols. 
+ */
+public class SegmentCompletionProtocolUtils {
+  public static SegmentCompletionProtocol.Response uploadSegmentWithFileUploadDownloadClient(
```

Comment:
can we just call this method `uploadSegment`? The file part is implied since the first argument is the `FileUploadDownloadClient`

Reply:
Or, maybe just inline the logic of this method inside the uploader. The logic for  a different uploader is going to be different anyway.

Reply:
Moved the method inside uploader.